python/ittapi/compat/__init__.py

```diff
@@ -0,0 +1,8 @@
+"""
+Python binding to NERSC implementation of Intel Instrumentation and Tracing Technology (ITT) API.
+"""
+from ittapi.native import pt_region_begin, pt_region_end
```

Comment:
`pt_region_begin` and `pt_region_end ` should also be available from the main ittapi module
need to add them into python/ittapi/__init__.py as well

Reply:
The `pt region` is implemented only to support the NERSC implementation. We can directly add the `pt region` APIs to the main `ittapi` module, and it will still work. 

It's just the general idea behind the main `ittapi` module is implement the bindings in pythonic way (e.g. decorators, context managers). So, we may need to add a few things to the main `ittapi` module to add the `pt region` and also support the general idea. 
I have the changes ready though, please have look at the following link: [Changes](https://github.com/mk-srivastava/ittapi/compare/dev_VASP-30439...mk-srivastava:ittapi:dev_VASP-30439_beta?expand=1). If you like I can push that changes to this PR.

Reply:
yes that's what I meant, we should add full support for pt_region to the main module as well. please add your changes to this PR

Reply:
I have added the support for `pt_region` to the main module as well as added the testcases for it too. Please have a look.